### ISSUE TITLE
Hexagonal crystals (R 3 -type spacegroups) 

### ISSUE THREAD

**Issue by JBGreisman** [project-gemmi/gemmi#26]
I ran into some confusion regarding the `R 3` family of spacegroups when writing tests related to spacegroups in `reciprocalspaceship`. I understand that spacegroups such as `R 3` can be represented in two different ways -- rhombohedral / hexagonal. However, when they are specified explicitly, it seems that `GroupOps.find_centering()` does not return a value for hexagonal representations:

```python
import gemmi

r3 = gemmi.SpaceGroup("R 3")
r3r = gemmi.SpaceGroup("R 3:R")
r3h = gemmi.SpaceGroup("R 3:H")

print(r3.operations().find_centering())  # prints '\x00'
print(r3r.operations().find_centering()) # prints 'P'
print(r3h.operations().find_centering()) # prints'\x00'
```

I am only showing this for `R 3:H`, but it is also true for the 6 other related spacegroups. It seems to me that there should be a returned centering in these cases, although I can understand not including a centering in the slightly ambiguous case of `R 3`. 

This also may be a misunderstanding of these spacegroups on my part, but the following symmetry operations are found for `R 3:H`: `[<gemmi.Op("x,y,z")>, <gemmi.Op("-y,x-y,z")>, <gemmi.Op("-x+y,-x,z")>]`. However, those symops match the rhombohedral setting here:   http://img.chem.ucl.ac.uk/sgp/large/146bz1.htm, and vice versa for `R 3:R`. I am uncertain whether these two are mixed-up in `gemmi`, or whether I am mixing them up in looking at the space group diagrams.

Thanks for your help!
Jack

**Comment by wojdyr** [comment]
Yes, find_centering() didn't work correctly. I just fixed it.
A couple months ago I changed the order of centering vectors for hexagonal groups, forgetting that find_centering() assumes that the operations are sorted.
Thanks for spotting the problem.

> This also may be a misunderstanding of these spacegroups on my part, but the following symmetry operations are found for `R 3:H`: `[<gemmi.Op("x,y,z")>, <gemmi.Op("-y,x-y,z")>, <gemmi.Op("-x+y,-x,z")>]`. However, those symops match the rhombohedral setting here: http://img.chem.ucl.ac.uk/sgp/large/146bz1.htm, and vice versa for `R 3:R`.

Indeed, `R 3:H` corresponds to [R3](http://img.chem.ucl.ac.uk/sgp/large/146bz1.htm) on the UCL website, under *Rhombohedral with Hexagonal axes*.
And `R 3:R` corresponds also to [R3](http://img.chem.ucl.ac.uk/sgp/large/146az1.htm) (*Trigonal*).
I suppose that's how it should be. I have a test that checks if the symmetry operations are the same as in cctbx (for all space group settings), so I hope there are no mistakes there.
